src/shaders/_prelude_fog.fragment.glsl

```diff
@@ -35,7 +36,11 @@ float fog_opacity(vec3 position) {
 }
 
 vec3 fog_apply(vec3 color, vec3 position) {
-    return gamma_mix(color, u_fog_color, fog_opacity(position));
+    return dither(gamma_mix(
```

Comment:
For reference, a worst case scenario where this is helpful with:

https://user-images.githubusercontent.com/7061573/112514998-1dff3a80-8d53-11eb-89bd-b99d8eb88d4e.mov

Also, the images from the PR description with increased brightness to see the behavior more clearly:

<img width="341" alt="112412732-3b44f200-8cdc-11eb-8823-36fa768ee9c9" src="https://user-images.githubusercontent.com/7061573/112515043-28b9cf80-8d53-11eb-9917-f68aff554063.png">
<img width="328" alt="112412717-37b16b00-8cdc-11eb-9307-b48b871a7adf" src="https://user-images.githubusercontent.com/7061573/112515047-29eafc80-8d53-11eb-9f8b-6a4c57662dad.png">